src/vcpkg/binarycaching.cpp

```diff
@@ -1007,7 +1015,8 @@ namespace
             {
                 auto&& action = *actions[idx];
                 const auto& abi = action.package_abi().value_or_exit(VCPKG_LINE_INFO);
-                if (m_tool->stat(make_object_path(m_prefix, abi)))
+                auto res = m_tool->stat(make_object_path(m_prefix, abi));
```

Comment:
If stat returns `CacheAvailability` then this can be:

```c++
auto maybe_res = m_tool->stat(make_object_path(m_prefix, abi));
if (auto res = maybe_res.get()) {
    cache_status[idx] = *res;
}
```